src/_P078_Eastron.ino

```diff
@@ -166,23 +157,63 @@ boolean Plugin_078(byte function, struct EventStruct *event, String& string)
-            options[i] = p078_getQueryString(i);
-          }
-          for (byte i = 0; i < P078_NR_OUTPUT_VALUES; ++i) {
-            const byte pconfigIndex = i + P078_QUERY1_CONFIG_POS;
-            sensorTypeHelper_loadOutputSelector(event, pconfigIndex, i, P078_NR_OUTPUT_OPTIONS, options);
-          }
-        }
+          int model = P078_MODEL;
 
+          if (model == 0) {
+              String options[P078_NR_OUTPUT_OPTIONS_SDM220_SDM120CT_SDM120];
+              for (int i = 0; i < P078_NR_OUTPUT_OPTIONS_SDM220_SDM120CT_SDM120; ++i) {
+                options[i] = p078_getQueryString(i,model);
+              }
+              for (byte i = 0; i < P078_NR_OUTPUT_VALUES; ++i) {
+              const byte pconfigIndex = i + P078_QUERY1_CONFIG_POS;
+              sensorTypeHelper_loadOutputSelector(event, pconfigIndex, i, P078_NR_OUTPUT_OPTIONS_SDM220_SDM120CT_SDM120, options);
+            }
+          } else if (model == 1) {
+              String options[P078_NR_OUTPUT_OPTIONS_SDM230];
+              for (int i = 0; i < P078_NR_OUTPUT_OPTIONS_SDM230; ++i) {
+                options[i] = p078_getQueryString(i,model);
+              }
+              for (byte i = 0; i < P078_NR_OUTPUT_VALUES; ++i) {
+              const byte pconfigIndex = i + P078_QUERY1_CONFIG_POS;
+              sensorTypeHelper_loadOutputSelector(event, pconfigIndex, i, P078_NR_OUTPUT_OPTIONS_SDM230, options);
+            } 
+          } else if (model == 2) {
+              String options[P078_NR_OUTPUT_OPTIONS_SDM72D];
+              for (int i = 0; i < P078_NR_OUTPUT_OPTIONS_SDM72D; ++i) {
+                options[i] = p078_getQueryString(i,model);
+              }
+              for (byte i = 0; i < P078_NR_OUTPUT_VALUES; ++i) {
+              const byte pconfigIndex = i + P078_QUERY1_CONFIG_POS;
+              sensorTypeHelper_loadOutputSelector(event, pconfigIndex, i, P078_NR_OUTPUT_OPTIONS_SDM72D, options);
+            }  
+          } else if (model == 3) {
+              String options[P078_NR_OUTPUT_OPTIONS_DDM18SD];
+              for (int i = 0; i < P078_NR_OUTPUT_OPTIONS_DDM18SD; ++i) {
+                options[i] = p078_getQueryString(i,model);
+              }
+              for (byte i = 0; i < P078_NR_OUTPUT_VALUES; ++i) {
+              const byte pconfigIndex = i + P078_QUERY1_CONFIG_POS;
+              sensorTypeHelper_loadOutputSelector(event, pconfigIndex, i, P078_NR_OUTPUT_OPTIONS_DDM18SD, options);
+            } 
+          } else if (model == 4) {
+              String options[P078_NR_OUTPUT_OPTIONS_SDM630];
+              for (int i = 0; i < P078_NR_OUTPUT_OPTIONS_SDM630; ++i) {
+                options[i] = p078_getQueryString(i,model);
+              }
+              for (byte i = 0; i < P078_NR_OUTPUT_VALUES; ++i) {
+              const byte pconfigIndex = i + P078_QUERY1_CONFIG_POS;
+              sensorTypeHelper_loadOutputSelector(event, pconfigIndex, i, P078_NR_OUTPUT_OPTIONS_SDM630, options);
+            }   
+          } 
```

Comment:
This is a lot of code duplication.
Better make the check for the model a `switch` statement and only set some variables.
After the switch statement you can then run the 2 for loops with the specific variables.

For the `String options[..]` array, I think you can declare the size with a run time variable, as long as you declare it inside a new scope (either a function call, or wrap it in {...}).
If that's not possible, just allocate the max. amount needed and only fill it to where you need it to be filled. The nr of options is one of the parameters of the sensorTypeHelper_loadOutputSelector.

But since the max. nr of options is quite big, you may want to split the execution of the `sensorTypeHelper_loadOutputSelector` and output the options immediately to the connected client.
Otherwise you may need to allocate close to 100 strings in memory just to serve an option list.
See for examples where `addSelector_Item` is used in the code.